### ISSUE TITLE
appium 1.18.1 and Xcode 12 beta 6 - Building scheme "iOS Framework" in CocoaAsyncSocket.xcodeproj failure

### ISSUE THREAD

**Issue by shaky** [appium/appium#14703]
## The problem

I'm trying to run my first test on Xcode 12 - Beta 6 but I get a Carthage failure when running **./Scripts/bootstrap.sh**

## Environment

* appium version: **1.18.1** and **v1.18.0-beta.1** (same results with both versions)
* Node.js version (unless using Appium.app|exe): **v12.18.3**
* Mobile platform/version under test: iOS 14.0 Simulator (Xcode 12, Beta 6 - **12A8189n**)
* macOS version: Catalina 10.15.6 (**19G2021**)
* Carthage version: **0.35.0**

## Details

- When I first ran **./Scripts/bootstrap.sh** I got the this error:
```

[debug] [WebDriverAgent] *** Building scheme "iOS Framework" in CocoaAsyncSocket.xcodeproj
[debug] [WebDriverAgent] Build Failed
[debug] [WebDriverAgent] 	Task failed with exit code 1:
[debug] [WebDriverAgent] 	/usr/bin/xcrun lipo -create /Users/testuser/Library/Caches/org.carthage.CarthageKit/DerivedData/12.0_12A8189n/CocoaAsyncSocket/72e0fa9e62d56e5bbb3f67e9cfd5aa85841735bc/Build/Intermediates.noindex/ArchiveIntermediates/iOS\ Framework/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/CocoaAsyncSocket.framework/CocoaAsyncSocket /Users/testuser/Library/Caches/org.carthage.CarthageKit/DerivedData/12.0_12A8189n/CocoaAsyncSocket/72e0fa9e62d56e5bbb3f67e9cfd5aa85841735bc/Build/Products/Release-iphonesimulator/CocoaAsyncSocket.framework/CocoaAsyncSocket -output /usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent/Carthage/Build/iOS/CocoaAsyncSocket.framework/CocoaAsyncSocket
[debug] [WebDriverAgent] This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/41/0dtvw_tx4v11gntm929t_lsm0000gs/T/carthage-xcodebuild.NH6ujt.log
[debug] [BaseDriver] Event 'wdaStartFailed' logged at 1599115027018 (07:37:07 GMT+0100 (British Summer Time))
[debug] [XCUITest] Unable to launch WebDriverAgent because of xcodebuild failure: Command 'carthage bootstrap --platform iOS\,tvOS' exited with code 1

```
**Full log here:**
[appium.log](https://github.com/appium/appium/files/5168897/appium.log)

-----------------------------------------------------------------------------

- Then I tried the solution mentioned here (which implies replacing **bootstrap.sh**) https://github.com/appium/appium/issues/14611#issuecomment-684877737 and then I got this error:

```
*** Building scheme "YYCache iOS" in YYCache.xcodeproj
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project /usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent/Carthage/Checkouts/YYCache/Framework/YYCache.xcodeproj -scheme YYCache\ iOS -configuration Release -derivedDataPath /Users/testuser/Library/Caches/org.carthage.CarthageKit/DerivedData/12.0_12A8189n/YYCache/1.1.0 -sdk iphonesimulator -destination platform=iOS\ Simulator,id=8FC4ADFD-9205-467B-A0CF-BEB43576CDC5 -destination-timeout 3 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES build (launched in /usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent/Carthage/Checkouts/YYCache)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/41/0dtvw_tx4v11gntm929t_lsm0000gs/T/carthage-xcodebuild.BRhXUe.log
```

**Full logs here:**
[bootstrap.log](https://github.com/appium/appium/files/5166960/bootstrap.log)
[carthage.log](https://github.com/appium/appium/files/5166961/carthage.log)

Please advice.

Thank you in advance!

**Comment by KazuCocoa** [comment]
Note: https://github.com/Carthage/Carthage/issues/3019

**Comment by HybridAU** [comment]
Thanks for that @KazuCocoa  I followed the workaround in the issue you linked to:

Created a file `/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-webdriveragent/tmp.xcconfig` with the contents

```
EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64=arm64 arm64e armv7 armv7s armv6 armv8
EXCLUDED_ARCHS=$(inherited) $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT))
```
then in that same directory ran

```
export XCODE_XCCONFIG_FILE=$PWD/tmp.xcconfig
carthage update --platform iOS --no-use-binaries --cache-builds
```

After that Appium built the WebDriverAgent with no issues and worked for me.

**Comment by shaky** [comment]
thanks @KazuCocoa. Will appium include a workaround in the next release?

@HybridAU I've tried that already but I get the second error now: 

*** Building scheme "YYCache iOS" in YYCache.xcodeproj
Build Failed
Task failed with exit code 65:
/usr/bin/xcrun xcodebuild -project /usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent/Carthage/Checkouts/YYCache/Framework/YYCache.xcodeproj -scheme YYCache\ iOS -configuration Release -derivedDataPath /Users/testuser/Library/Caches/org.carthage.CarthageKit/DerivedData/12.0_12A8189n/YYCache/1.1.0 -sdk iphonesimulator -destination platform=iOS\ Simulator,id=8FC4ADFD-9205-467B-A0CF-BEB43576CDC5 -destination-timeout 3 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES build (launched in /usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent/Carthage/Checkouts/YYCache)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/41/0dtvw_tx4v11gntm929t_lsm0000gs/T/carthage-xcodebuild.BRhXUe.log

**Comment by shaky** [comment]
@HybridAU what's the actual appium version you're using? have you done a fresh install or you installed the new version of appium on top of an old one?

Also, what's the Xcode version you're using?

Thanks!

**Comment by HybridAU** [comment]
@shaky I'm using Appium 1.18.1 installed with brew and Xcode 12 beta 6 installed with xcversion.

With Appium I had one version that was installed with npm and another that was installed by downloading the .dmg from GitHub and was having issues so I tried to remove everything and do a fresh install. I think it was a fresh install but there might have been some artifacts left from before.

**Comment by shaky** [comment]
Thanks for the details @HybridAU. 

Indeed, you might have some artifacts left behind that helped you make it work. 
I'm using a fresh Macbook running Catalina 10.15.6 + appium 1.18.1 + Xcode 12, Beta 6 - 12A8189n

----------------------------------------------------------------------------------------------------

I finally managed to make it work using the following steps:

Deleted everything inside ~/Library/Developer/Xcode/DerivedData
`brew uninstall carthage`
`npm install -g appium --chromedriver-skip-install`
`sudo chown -R $USER /usr/local/lib`
`cd /usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent`
`mkdir -p Resources/WebDriverAgent.bundle`
based on https://github.com/appium/appium/issues/14611#issuecomment-677458929 I replaced the content of `/Scripts/bootstrap.sh with`

```
xcconfig=$(mktemp /tmp/static.xcconfig.XXXXXX)
trap 'rm -f "$xcconfig"' INT TERM HUP EXIT
echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_12A8189h = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200 = $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_$(XCODE_PRODUCT_BUILD_VERSION))' >> $xcconfig
echo 'EXCLUDED_ARCHS = $(inherited) $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)_NATIVE_ARCH_64_BIT$(NATIVE_ARCH_64_BIT)_XCODE$(XCODE_VERSION_MAJOR))' >> $xcconfig

export XCODE_XCCONFIG_FILE="$xcconfig"
```

then ran
`carthage bootstrap $USE_SSH --platform iOS $NO_USE_BINARIES` => bootstrap ran without errors

I started the `appium` server(CLI) and tried to run a test => Failure
appium was complaining that is unable to find the `WebDriverAgentRunner-Runner.app` inside ~/Library/Developer/Xcode/DerivedData/WebDriverAgent-ciegwgvxzxdrqthilmrmczmqvrgu/Build/Debug-iphonesimulator
so I manually copied `WebDriverAgentRunner-Runner.app` from `/usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent` to `~/Library/Developer/Xcode/DerivedData/WebDriverAgent-ciegwgvxzxdrqthilmrmczmqvrgu/Build/Debug-iphonesimulator`

Now everything seems fine :)

I moved the whole `WebDriverAgent-ciegwgvxzxdrqthilmrmczmqvrgu` folder to my project and started started using `derivedDataPath` and `usePrebuiltWDA`

**Comment by mykola-mokhnach** [comment]
It looks like there was an issue with the npm package, where one of the scripts was not included into the dist. the fix for it has been published today in appium@beta

**Comment by shaky** [comment]
Thanks @mykola-mokhnach !
I'll try it out.

**Comment by Ichhakhemka** [comment]
I am still getting "Could not build module 'CocoaAsyncSocket' " error while building project. so fix for this will come in v1.18.2 for appium?
Currently using Appium 1.1.8.1 with xcode 12 beta 6

**Comment by Ichhakhemka** [comment]
@mykola-mokhnach  what is the release date for this new fix? I am also facing same issue "Could not build module 'CocoaAsyncSocket' " and "Use of '@import' in framework header is discouraged, including this header requires -fmodules" when building project.

Using Appium 1.18.1, Xcode 12 beta 6
